### PR TITLE
Python: supress non-useful results (w/ tests) for iter str/seq query 

### PR DESCRIPTION
**Pull request by RasmusWL**
Based on #3212, so that should be merged first.

_I can't create a PR on QL with #3212 as the base branch, only for my own repo, which is not useful :neutral_face:_

### CODE REVIEW

**Comment by RasmusWL** [comment]
Hmm, now that the 3 first commits are in `master`, I had actually hoped that they wouldn't be included in the list of commits :confused:

I'll give it a few minutes, and otherwise try to merge in master

File: python/ql/test/3/query-tests/Statements/iter/supress_uses_from_tests.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# This example illustrates that all valid results are not useful.
```
Comment (tausbn) [inline python/ql/test/3/query-tests/Statements/iter/supress_uses_from_tests.py:1]: I interpret this as saying `forall(Result r | r.valid() implies not r.useful())`, which sounds a bit odd to me.

Did you intend to say `not forall(Result r | r.valid() implies r.useful())`?
(i.e. `not all valid results are useful`?)

Comment (RasmusWL) [reply python/ql/test/3/query-tests/Statements/iter/supress_uses_from_tests.py:1]: :smile: yes

**Comment by tausbn** [comment]
This should be safe to merge now.

**Pull request merged**